{
  "repository": {
    "name": "vert-x3/vertx-mysql-postgresql-client",
    "githubId": 27534295,
    "url": "https://github.com/vert-x3/vertx-mysql-postgresql-client",
    "isFork": false,
    "original": {
      "name": "",
      "githubId": 0,
      "url": ""
    },
    "isPullRequest": true,
    "pullRequestId": 120
  },
  "builds": {
    "buggyBuild": {
      "id": 405525214,
      "url": "http://travis-ci.org/vert-x3/vertx-mysql-postgresql-client/builds/405525214",
      "date": "Jul 18, 2018 10:20:40 PM"
    }
  },
  "reproductionBuggyBuild": {
    "reproductionDateBeginning": "Jul 18, 2018 10:20:53 PM",
    "reproductionDateEnd": "Jul 18, 2018 10:29:22 PM",
    "totalDuration": 504,
    "processDurations": {
      "cloning": {
        "nbSteps": 1,
        "totalDuration": 4,
        "stepNames": [
          "CloneRepository"
        ],
        "stepDurations": [
          4
        ]
      },
      "building": {
        "nbSteps": 2,
        "totalDuration": 33,
        "stepNames": [
          "CheckoutBuggyBuild",
          "BuildProject"
        ],
        "stepDurations": [
          1,
          32
        ]
      },
      "testing": {
        "nbSteps": 1,
        "totalDuration": 445,
        "stepNames": [
          "TestProject"
        ],
        "stepDurations": [
          445
        ]
      }
    },
    "projectRootPomPath": "./workspace/vert-x3/vertx-mysql-postgresql-client/405525214/pom.xml"
  },
  "patchDiff": {
    "files": {
      "numberAdded": 0,
      "numberChanged": 0,
      "numberDeleted": 0
    },
    "lines": {
      "numberAdded": 0,
      "numberDeleted": 0
    }
  },
  "commits": {
    "buggyBuild": {
      "repoName": "vert-x3/vertx-mysql-postgresql-client",
      "branchName": "master",
      "sha": "95612741b616b011118ba01960a2f9dc6ca36c1b",
      "url": "http://github.com/vert-x3/vertx-mysql-postgresql-client/commit/95612741b616b011118ba01960a2f9dc6ca36c1b",
      "date": "Jul 18, 2018 10:18:20 PM"
    },
    "buggyBuildForkRepo": {
      "repoName": "vert-x3/vertx-mysql-postgresql-client",
      "branchName": "fix-postgresql-ssl-test",
      "sha": "4b2453b9cce922149c00327c8c16cce3451fc4be",
      "url": "https://github.com/vert-x3/vertx-mysql-postgresql-client/commit/4b2453b9cce922149c00327c8c16cce3451fc4be",
      "date": "Jul 18, 2018 10:18:20 PM"
    },
    "buggyBuildBaseRepo": {
      "repoName": "vert-x3/vertx-mysql-postgresql-client",
      "branchName": "master",
      "sha": "48cfb24d253c072d60cc9f0d411cf8be6907ca09",
      "url": "https://github.com/vert-x3/vertx-mysql-postgresql-client/commit/48cfb24d253c072d60cc9f0d411cf8be6907ca09",
      "date": "Jul 18, 2018 1:25:17 AM"
    }
  },
  "projectMetrics": {
    "numberModules": 1,
    "numberSourceFiles": 22,
    "numberTestFiles": 16,
    "numberLibrariesFailingModule": 47
  },
  "tests": {
    "failingModule": "/root/workspace/vert-x3/vertx-mysql-postgresql-client/405525214",
    "overallMetrics": {
      "numberRunning": 53,
      "numberPassing": 13,
      "numberFailing": 2,
      "numberErroring": 38,
      "numberSkipping": 2,
      "failures": [
        {
          "failureName": "java.lang.AssertionError",
          "isError": false,
          "occurrences": 2
        },
        {
          "failureName": "com.wix.mysql.exceptions.MissingDependencyException",
          "isError": true,
          "occurrences": 2
        },
        {
          "failureName": "java.util.concurrent.TimeoutException",
          "isError": true,
          "occurrences": 2
        },
        {
          "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
          "isError": true,
          "occurrences": 34
        }
      ]
    },
    "failingClasses": [
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLTest",
        "numberRunning": 5,
        "numberPassing": 0,
        "numberFailing": 0,
        "numberErroring": 5,
        "numberSkipping": 0
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLConfigurationTest",
        "numberRunning": 2,
        "numberPassing": 0,
        "numberFailing": 0,
        "numberErroring": 2,
        "numberSkipping": 1
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLSslConfigurationTest",
        "numberRunning": 5,
        "numberPassing": 1,
        "numberFailing": 2,
        "numberErroring": 2,
        "numberSkipping": 1
      },
      {
        "testClass": "io.vertx.ext.asyncsql.MySQLClientTest",
        "numberRunning": 1,
        "numberPassing": 0,
        "numberFailing": 0,
        "numberErroring": 1,
        "numberSkipping": 0
      },
      {
        "testClass": "io.vertx.ext.asyncsql.MySQLConfigurationTest",
        "numberRunning": 1,
        "numberPassing": 0,
        "numberFailing": 0,
        "numberErroring": 1,
        "numberSkipping": 0
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "numberRunning": 29,
        "numberPassing": 2,
        "numberFailing": 0,
        "numberErroring": 27,
        "numberSkipping": 0
      }
    ],
    "failureDetails": [
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testSimpleSelect",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLSslConfigurationTest",
        "testMethod": "testPreferSslConfiguration",
        "failureName": "java.lang.AssertionError",
        "detail": "Expected true",
        "isError": false
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testTwoTransactionsAfterEachOther",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLConfigurationTest",
        "testMethod": "testCharset",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testInsertion",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testMultipleConnections",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLSslConfigurationTest",
        "testMethod": "testQueryTimeout",
        "failureName": "java.util.concurrent.TimeoutException",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLTest",
        "testMethod": "someTest",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testCommitWhenNotInTransaction",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testByteA2Fields",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testInvalidInsertStatement",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testDecimalFields",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testLongStream",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testSimpleConnection",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLSslConfigurationTest",
        "testMethod": "testCharset",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:54321",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testSetAutocommitWhileInTransaction",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testByteA1Fields",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testUsingUUIDsInTables",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testColumnsFromStream",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testUpdatingRow",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.MySQLConfigurationTest",
        "testMethod": "io.vertx.ext.asyncsql.MySQLConfigurationTest",
        "failureName": "com.wix.mysql.exceptions.MissingDependencyException",
        "detail": "System library \u0027libaio.so.1\u0027 missing. Please install it via system package manager, ex. \u0027sudo apt-get install libaio1\u0027.\nFor details see: http://bugs.mysql.com/bug.php?id\u003d60544",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLSslConfigurationTest",
        "testMethod": "testCorrectSslConfiguration",
        "failureName": "java.lang.AssertionError",
        "detail": "Expected true",
        "isError": false
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testSelectionOfNullValues",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testRollingBackWhenNotInTransaction",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLTest",
        "testMethod": "queryTypeTimestampWithTimezoneTest",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testUnhandledExceptionInHandlerResultSet",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testDateValueSelection",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testInstant",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLTest",
        "testMethod": "queryArrayTypeTest",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testByteA3Fields",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLTest",
        "testMethod": "queryTypeTimestampWithoutTimezoneTest",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testUnhandledExceptionInHandlerUpdateResult",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testSimpleStream",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testRollback",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testInsertedIds",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.MySQLClientTest",
        "testMethod": "io.vertx.ext.asyncsql.MySQLClientTest",
        "failureName": "com.wix.mysql.exceptions.MissingDependencyException",
        "detail": "System library \u0027libaio.so.1\u0027 missing. Please install it via system package manager, ex. \u0027sudo apt-get install libaio1\u0027.\nFor details see: http://bugs.mysql.com/bug.php?id\u003d60544",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testTimeColumn",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLClientTest",
        "testMethod": "testUnhandledExceptionInHandlerRowStream",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLTest",
        "testMethod": "testUpdatingNumericField",
        "failureName": "io.netty.channel.AbstractChannel$AnnotatedConnectException",
        "detail": "Connection refused: localhost/127.0.0.1:5432",
        "isError": true
      },
      {
        "testClass": "io.vertx.ext.asyncsql.PostgreSQLConfigurationTest",
        "testMethod": "testQueryTimeout",
        "failureName": "java.util.concurrent.TimeoutException",
        "isError": true
      }
    ]
  },
  "type": "only_fail"
}